File: 1402-count-square-submatrices-with-all-ones/1402-count-square-submatrices-with-all-ones.py
class Solution(object):
    def countSquares(self, matrix):
        n, m, ans = len(matrix), len(matrix[0]), 0
        for i in range(n):
            for j in range(m):
                if i != 0 and j != 0 and matrix[i][j] != 0:
                    matrix[i][j] = min(min(matrix[i-1][j], matrix[i][j-1]), matrix[i-1][j-1]) + 1
                ans += matrix[i][j]
        
        return ans
        

File: 1402-count-square-submatrices-with-all-ones/1402-count-square-submatrices-with-all-ones.cpp
class Solution {
public:
    int countSquares(vector<vector<int>>& matrix) {
        int n = matrix.size(), m = matrix[0].size(), ans = 0;
        for(int i = 0; i<n; i++) {
            for(int j = 0; j<m; j++) {
                if(i && j && matrix[i][j]) {
                    matrix[i][j] = min({matrix[i-1][j], matrix[i][j-1], matrix[i-1][j-1]}) + 1;
                }
                ans += matrix[i][j];
            }
        }

        return ans;
    }
};